src/main/java/net/sf/jabref/logic/l10n/Localization.java

```diff
@@ -17,28 +19,37 @@
+        Locale locale = new Locale(language);
 
-        messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
-        menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
-        intMessages = ResourceBundle.getBundle(INTEGRITY_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+        try {
+            messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            intMessages = ResourceBundle.getBundle(INTEGRITY_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
 
-        // these checks are required as when the requested resource bundle is NOT found, the default locale is used as a fallback silently.
-        if(!messages.getLocale().equals(locale)) {
-            LOGGER.warn("tried loading <" + RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back on default locale <" + Locale.getDefault() + ">");
-        }
+            // silent fallback to system locale when bundle is not found
+            if(!messages.getLocale().equals(locale)) {
+                LOGGER.warn("Tried loading <" + RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back to default locale <" + defaultLocale + ">");
+            }
 
-        if(!menuTitles.getLocale().equals(locale)) {
-            LOGGER.warn("tried loading <" + MENU_RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back on default locale <" + Locale.getDefault() + ">");
-        }
+            if(!menuTitles.getLocale().equals(locale)) {
+                LOGGER.warn("Tried loading <" + MENU_RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back to default locale <" + defaultLocale + ">");
+            }
 
-        if(!intMessages.getLocale().equals(locale)) {
-            LOGGER.warn("tried loading <" + INTEGRITY_RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back on default locale <" + Locale.getDefault() + ">");
+            if(!intMessages.getLocale().equals(locale)) {
+                LOGGER.warn("Tried loading <" + INTEGRITY_RESOURCE_PREFIX + "> for locale <" + locale + "> but had to fall back to default locale <" + defaultLocale + ">");
+            }
+        } catch(MissingResourceException e) {
+            LOGGER.warn("Fallback to system locale <" + defaultLocale + "> failed, using locale <en> instead");
+
+            locale = new Locale("en");
+            messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            intMessages = ResourceBundle.getBundle(INTEGRITY_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+        } finally {
+            Locale.setDefault(locale);
```

Comment:
Are these two lines really needed? What do they change exactly? maybe we should add a clarifying comment!